bleak/__init__.py

```diff
@@ -204,6 +206,53 @@ def set_scanning_filter(self, **kwargs):
         )
         self._backend.set_scanning_filter(**kwargs)
 
+    def __aiter__(self) -> AsyncIterator[Tuple[BLEDevice, AdvertisementData]]:
+        self._backend.new_devices = asyncio.Queue()
+        return self
+
+    async def __anext__(self) -> Tuple[BLEDevice, AdvertisementData]:
+        return await self._backend.new_devices.get()
```

Comment:
Not sure if implementing this is even required... It saved 10 chars of typing (`.devices()`), but cannot have the timeout or return type functionality. Additionally, `async for .. in scanner:` might be less intuitive than `async for .. in scanner.devices():`

Reply:
I would rather not make the `BleakScanner` class itself iterable.

Reply:
Done, methods removed.